packages/json/lib/commands/TYPE.ts

```diff
@@ -18,8 +18,8 @@ export default {
   },
   transformReply: {
     2: undefined as unknown as () => NullReply | BlobStringReply | ArrayReply<BlobStringReply>,
-    // TODO: ?!??!
-    3: undefined as unknown as () => any
-  }
+    3: undefined as unknown as () => ArrayReply<NullReply | BlobStringReply | ArrayReply<BlobStringReply>>
```

Comment:
I'm not too sure if this is right..

Reply:
in resp3, the same reply that is possible in resp2 is returned in resp3, just wrapped in an array.

```rust
pub fn json_type<M: Manager>(manager: M, ctx: &Context, args: Vec<RedisString>) -> RedisResult {
    let mut args = args.into_iter().skip(1);
    let key = args.next_arg()?;
    let path = Path::new(args.next_str().unwrap_or(default_path()));

    let key = manager.open_key_read(ctx, &key)?;

    let value = if path.is_legacy() {
        json_type_legacy::<M>(&key, path.get_path())?
    } else {
        json_type_impl::<M>(&key, path.get_path())?
    };

    // Check context flags to see if RESP3 is enabled and return the appropriate result
    if is_resp3(ctx) {
        Ok(vec![value].into())
    } else {
        Ok(value)
    }
}
```

so if one is ok with the resp2, response, the resp3 response looks right to me.

the problem is really in me messing up the conversion in v4, we have

```
export declare function transformReply(): string | null | Array<string | null>;
```

so what we need is the ArrayReply<BlobStringReply | NullReply>

Reply:
made the change I mentioned now